lineman/app/services/form_service.coffee

```diff
@@ -13,23 +13,49 @@ angular.module('loomioApp').factory 'FormService', ($rootScope, FlashService, $f
       422: 'unprocessableEntity'
       500: 'internalServerError'
 
+    prepare = (scope, model, options) ->
+      FlashService.loading()
+      scope.isDisabled = true
+      model.setErrors
+
+    success = (scope, model, options) ->
+      (response) ->
+        FlashService.dismiss()
+        FlashService.success options.flashSuccess, options.flashOptions if options.flashSuccess?
+        scope.$close()                                                  if typeof scope.$close is 'function'
+        options.successCallback(response)                               if typeof options.successCallback is 'function'
+
+    failure = (scope, model, options) ->
+      (response) ->
+        FlashService.dismiss()
+        model.setErrors response.data.errors                            if response.status == 422
+        $rootScope.$broadcast errorTypes[response.status] or 'unknownError',
+          model: model
+          response: response
+
+    cleanup = (scope) ->
```

Comment:
Should have happened already when you submit the form, I think; that's what the `setErrors()` you mentioned above is trying to do. (I think it's not so good to potentially sit there and look at a validation error while the form is processing)